solution/index.js

```diff
@@ -0,0 +1,446 @@
+ if(!localStorage.tasks){
+    localStorage.setItem('tasks', JSON.stringify({
+        "todo": [],
+        "in-progress": [],
+        "done": []
+          })
+        );
+ }
+
+ //localStorage.clear(); // clears the local storage
+
+    console.log(localStorage.tasks);
+  // initilizes the local storgae object for updates
+    if(typeof(localStorage.tasks) === 'string'){
+        var localStorageObjectForUpdate = JSON.parse(localStorage.tasks);
+    }else{   
+        var localStorageObjectForUpdate = localStorage.tasks;
+    }
+
+
+document.body.addEventListener('mouseover' ,addHoverReplace);
```

Comment:
I would add this event listener to every task individually, so you already have the element in hand.
You don't always have to use event delegation - that's just a pattern some people use sometimes...